addons/web/static/src/js/views/basic/basic_renderer.js

```diff
@@ -323,7 +323,18 @@ var BasicRenderer = AbstractRenderer.extend(WidgetAdapterMixin, {
             element.$el.toggleClass("o_readonly_modifier", !!modifiers.readonly);
             element.$el.toggleClass("o_required_modifier", !!modifiers.required);
 
-            if (element.widget && element.widget.updateModifiersValue) {
+            if (element.widget instanceof owl.Component) {
+                // props have changed but we do not want to re-render the component
+                // because we've just updated the rendering when toggling css classes.
+                const props = element.widget.props;
+                if (props.options.attrs) {
+                    Object.assign(props.options.attrs, modifiers);
+                } else {
+                    const viewType = props.options.viewType || 'default';
+                    const fieldsInfo = props.record.fieldsInfo[viewType];
+                    Object.assign(fieldsInfo[props.fieldName].modifiers, modifiers);
```

Comment:
red flag here, the fieldsInfo stuff is tricky, not sure you should override the modifiers there. You should triple check that, and if you want we can talk about it